lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -1343,18 +1373,26 @@ class DedupPass : public DedupBase<DedupPass> {
     // Calculate module information parallelly.
     mlir::parallelFor(context, 0, modules.size(), [&](unsigned idx) {
       auto module = modules[idx];
+      AnnotationSet annotations(module);
       // If the module is marked with NoDedup, just skip it.
-      if (AnnotationSet(module).hasAnnotation(noDedupClass))
+      if (annotations.hasAnnotation(noDedupClass))
         return;
+
       // If the module has input RefType ports, also skip it.
       if (llvm::any_of(module.getPorts(), [&](PortInfo port) {
             return type_isa<RefType>(port.type) && port.isInput();
           }))
         return;
 
       StructuralHasher hasher(hasherConstants);
+      auto dedupGroup =
+          annotations.hasAnnotation(dedupGroupAnnoClass)
+              ? std::optional{annotations.getAnnotation(dedupGroupAnnoClass)
+                                  .getMember<StringAttr>("group")}
+              : std::nullopt;
```

Comment:
StringAttr is nullable so can remove std::optinal? 
```suggestion
          annotations.hasAnnotation(dedupGroupAnnoClass)
              ? annotations.getAnnotation(dedupGroupAnnoClass)
                                  .getMember<StringAttr>("group")
              : StringAttr();
```